### PR TITLE
docs: small changes

### LINKED ISSUE #939: [Bug- Documentation] unsafeHttpWhitelist option missing
Hi, I am trying to add Yarn 2 at work and there is a need for adding a private registry.

From looking the documentation `npmRegistryServer` option can be used to assign a private registry link but on running `yarn` it says its unsafe as it uses the `http` protocol since its in the intranet.

I couldn't find the `unsafeHttpWhitelist` option to add the domain there for whitelisting when the url is unsafe.

Lucky enough `yarn config` showed me the option but no documentation about it. 

This helped to understand what was going on https://github.com/yarnpkg/berry/pull/416

So this is merely a suggestion to add it to the documentation.

For anybody looking into adding a private registry you need to add two options in `.yarnrc.yml` file

```yaml
# .yarnrc.yml
npmRegistryServer: "http://artifactory.com/npm/"
unsafeHttpWhitelist: ["artifactory.com"]
```

### LINKED ISSUE #946: [Documentation] Add notes for React Native
- [x ] I'd be willing to implement this feature
- [ ] This feature can already be implemented through a plugin

**Describe the user story**

Yarn v2 installation documents doesn't mention that `React Native` projects have to use `nodeLinker: node-modules`. It is mentioned in migration guide. But it doesn't clearly have a `React Native` title. I noticed it when I tried to run `pod install`.

**Describe the solution you'd like**

Node linker usage should be in install instractions.

**Describe the drawbacks of your solution**

None.

**Describe alternatives you've considered**

None.

**Additional context**

None.

### LINKED ISSUE #1029: [Bug] Wrong link in the documentation
- [ ] I'd be willing to implement a fix

There is a link to `#npmServer` section, but there is no such section in docs

**Additional context**

Here is the line with an error https://github.com/yarnpkg/berry/blob/572775e2adc8b342b33c349218c1a9b290e31bee/packages/gatsby/src/pages/configuration/yarnrc.js#L178

### LINKED ISSUE #1085: [Bug] Documentation: yarn add pkg@<git repo> syntax missing from CLI / website
- [x] I'd be willing to implement a fix

**Describe the bug**

`yarn add` supports adding git repos as dependencies (as opposed to registered npm packages). I'm not 100% certain of what's supported, but it seems that yarn's syntax differs from npm's:

```sh
# npm
$ npm install https://github.com/indexzero/forever

# yarn
$ yarn add forever@https://github.com/indexzero/forever
```

It would be really nice to have CLI help documentation that covers this. I'd be willing to prepare a PR it, but since I don't know what's possible and what's not, I wouldn't exactly say that I'm qualified.

### LINKED ISSUE #1094: [Bug] Docs say deferred versions are stored in .yarn/releases; are actually stored in .yarn/versions
- [ ] I'd be willing to implement a fix

**Describe the bug**

Docs here: https://yarnpkg.com/features/release-workflow
Say that deferred versions (`yarn version -d`) are stored in .yarn/releases but it looks like they actually get stored in .yarn/versions.

I omitted the rest of the issue template because this looks like a typo in the docs; not a runtime bug.

### PR DESCRIPTION
**Pull request by paul-soporan**
### What's the problem this PR addresses?

There are a few missing configuration options in the docs and other small things to fix / change. I am trying to find them now and fix / change them. This PR is in progress.

I want this PR to serve as a "Mega PR" used to fix / change the small things that don't deserve their own PRs.

#### Changes:
- add `deferredVersionFolder` to the config page and change `./.yarn/releases` to `./.yarn/versions`. Fixes #1094.
- add `enableHyperlinks` to the config page
- add `preferUnplugged` to the manifest config page
- add `initLicense` to the config page
- add `initScope` to the config page
- add `initVersion` to the config page
- improve `nodeLinker` description and highlight `pnpMode` values
- add `ignorePath` to the config page
- add `installStatePath` to the config page
- add `enableAbsoluteVirtuals` to the config page
- add `progressBarStyle` to the config page
- add `defaultLanguageName` to the config page
- improve descriptions of `unsafeHttpWhitelist` and `npmRegistryServer`. Fixes #939.
- add `type` option to the manifest config page
- change `npmServer` to `npmRegistries` in `npmAuthToken` description. Fixes #1029.
- improve the `node-modules` plugin documentation on the Migration page
- link the `node-modules` plugin documentation from the PnP page, the Installation page, and the Usage page. Fixes #946.
- add flow, react-native, and vsce as incompatible tools to the compatibility table
- add `Compatibility Table` heading on the PnP page
- add `files` option to the manifest config page
- document adding github repositories with `yarn add`. Closes #1085.
- document pnp `loose` mode
- add table of contents to the migration page
- add missing default plugins to readme
- make `Node API` from the readme a link to the api page

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,16 +95,20 @@ Note that no other command is needed! Given that our dependencies are checked-in
...
 - [★ plugin-http](packages/plugin-http) adds support for using straight URL references as dependencies (tgz archives only).
 - [★ plugin-init](packages/plugin-init) adds support for the [`yarn init`](https://yarnpkg.com/cli/init) command.
 - [★ plugin-link](packages/plugin-link) adds support for using [`link:` and `portal:`](https://yarnpkg.com/features/protocols#whats-the-difference-between-link-and-portal) references as dependencies.
+- [★ plugin-node-modules](packages/plugin-node-modules) adds support for installing packages through a `node_modules` folder.
 - [★ plugin-npm](packages/plugin-npm) adds support for using [semver ranges](https://semver.org) as dependencies, resolving them to an NPM-like registry.
 - [★ plugin-npm-cli](packages/plugin-npm-cli) adds support for the NPM-specific commands ([`yarn npm info`](https://yarnpkg.com/cli/npm/info), [`yarn npm login`](https://yarnpkg.com/cli/npm/login), [`yarn npm publish`](https://yarnpkg.com/cli/npm/publish), ...).
 - [★ plugin-pack](packages/plugin-pack) adds support for the [`yarn pack`](https://yarnpkg.com/cli/pack) command.
+- [★ plugin-patch](packages/plugin-patch) adds support for the patch: protocol.
```
Comment (arcanis) [inline README.md]:
```suggestion
- [★ plugin-patch](packages/plugin-patch) adds support for the `patch:` protocol.
```

File: packages/gatsby/content/advanced/migration.md [hunk 2]
```diff
@@ -4,11 +4,31 @@ path: /advanced/migration
...
+  - [Setup your IDE for PnP support](#setup-your-ide-for-pnp-support)
+  - [Update your configuration to the new settings](#update-your-configuration-to-the-new-settings)
+  - [Don't use `.npmrc` files](#dont-use-npmrc-files)
+  - [Take a look at our end-to-end tests](#take-a-look-at-our-end-to-end-tests)
+  - [Don't use `bundleDependencies`](#dont-use-bundledependencies)
+  - [If required: enable the `node-modules` plugin](#if-required-enable-the-node-modules-plugin)
+- [Troubleshooting](#troubleshooting)
+  - [`Cannot find module [...]`](#cannot-find-module-)
+    - [Make sure you use `resolve@1.9+`](#make-sure-you-use-resolve19)
+  - [`A package is trying to access another package [...]`](#a-package-is-trying-to-access-another-package-)
```
Comment (arcanis) [inline packages/gatsby/content/advanced/migration.md]: Great idea! Do you know if there's a Gatsby plugin that would avoid us from having to synchronize this list? We use another table of content in the FaQ, could make sense to factorize that (we can do it in a separate PR).

File: packages/gatsby/content/advanced/migration.md [hunk 3]
```diff
@@ -124,18 +144,22 @@ So how to replace them? There are different ways:
...
 Despite our best efforts some tools don't work at all under Plug'n'Play environments, and we don't have the resources to update them ourselves. There are only two notorious ones on our list: Flow, and React Native.
 
-In such a radical case, you can enable the [`node-modules` plugin](https://github.com/yarnpkg/berry/tree/master/packages/plugin-node-modules) by adding the following into your local `.yarnrc.yml` file before running a fresh `yarn install`:
+In such a radical case, you can enable the built-in [`node-modules` plugin](https://github.com/yarnpkg/berry/tree/master/packages/plugin-node-modules) by adding the following into your local [`.yarnrc.yml`](/configuration/yarnrc) file before running a fresh `yarn install`:
 
 ```yaml
 nodeLinker: node-modules
 ```
 
 This will cause Yarn to install the project just like Yarn 1 used to, by copying the packages into various `node_modules` folders. Remember that this plugin is currently very experimental and we expect it to improve over time.
```
Comment (arcanis) [inline packages/gatsby/content/advanced/migration.md]:
```suggestion
This will cause Yarn to install the project just like Yarn 1 used to, by copying the packages into various `node_modules` folders.
```

File: packages/gatsby/content/features/plugnplay.md [hunk 4]
```diff
@@ -38,29 +38,56 @@ This approach has various benefits:
 
 - Faster application startup, because the Node resolution doesn't have to iterate over the filesystem hierarchy nearly as much as before (and soon won't have to do it at all!).
 
+## PnP `loose` mode
+
+PnP `strict` mode breaks packages that rely on hoisting.
```
Comment (arcanis) [inline packages/gatsby/content/features/plugnplay.md]:
```suggestion
Because the hoisting heuristics aren't standardized and predictable, PnP operating under strict mode will prevent packages to require dependencies that they don't explicitly list (even if one of their others dependencies happens to depend on it). This may cause issues with some packages.
```

File: packages/gatsby/content/features/plugnplay.md [hunk 5]
```diff
@@ -38,29 +38,56 @@ This approach has various benefits:
...
+
+The `loose` mode will cause Yarn to call the `node-modules` hoister to generate the list of packages that would have been hoisted to the top-level in a typical `node_modules` install. It follows the exact same implementation as the true hoister used by the [`node-modules` plugin](https://github.com/yarnpkg/berry/tree/master/packages/plugin-node-modules). This set of packages will be stored in a "fallback pool" that will be made available to the transitive dependencies, following the policy defined in [pnpFallbackMode](/configuration/yarnrc#pnpFallbackMode). Accessing the fallback pool will now generate a warning through [process.emitWarning](https://nodejs.org/api/process.html#process_process_emitwarning_warning_type_code_ctor).
```
Comment (arcanis) [inline packages/gatsby/content/features/plugnplay.md]:
```suggestion
To address this problem, Yarn ships with a "loose" mode which will cause the PnP linker to work in tandem with the `node-modules` hoister - we will first generate the list of packages that would have been hoisted to the top-level in a typical `node_modules` install, then remember this list as what we call the "fallback pool".

> Note that because the loose mode directly calls the `node-modules` hoister, it follows the exact same implementation as the true algorithm used by the [`node-modules` linker](https://github.com/yarnpkg/berry/tree/master/packages/plugin-node-modules)!

At runtime, packages that require unlisted dependencies will still be allowed to access them if any version of the dependency ended up in the fallback pool (which packages exactly are allowed to rely on the fallback pool can be tweaked with [pnpFallbackMode](/configuration/yarnrc#pnpFallbackMode)).

Note that the content of the fallback pool is undetermined - should a dependency tree contains multiple versions of a same package, there's no telling which one will be hoisted to the top-level! For this reason, a package accessing the fallback pool will still generate a warning (via the [process.emitWarning](https://nodejs.org/api/process.html#process_process_emitwarning_warning_type_code_ctor) API).
```

File: packages/gatsby/content/features/plugnplay.md [hunk 6]
```diff
@@ -38,29 +38,56 @@ This approach has various benefits:
...
+| VSCode Extension Manager (vsce) | Incompatible    | Use the [vsce-yarn-patch](https://www.npmjs.com/package/vsce-yarn-patch) fork with the `node-modules` plugin enabled. The fork is required until [microsoft/vscode-vsce#379](https://github.com/microsoft/vscode-vsce/pull/379) is merged, as `vsce` currently uses the removed `yarn list` command |
 
 This list is kept up-to-date based on the latest release we've published starting from the v2. In case you notice something off in your own project please try to upgrade Yarn and the problematic package first, then feel free to file an issue. And maybe a PR? 😊
+
+---
+
+If a tool is incompatible with PnP, you can enable the [`node-modules` plugin](https://github.com/yarnpkg/berry/tree/master/packages/plugin-node-modules).
+
+[Instructions](/advanced/migration#if-required-enable-the-node-modules-plugin).
```
Comment (arcanis) [inline packages/gatsby/content/features/plugnplay.md]:
```suggestion
**Important:** If a tool is incompatible with PnP, you can still enable the [`node-modules` plugin](https://github.com/yarnpkg/berry/tree/master/packages/plugin-node-modules). Just follow the [instructions](/advanced/migration#if-required-enable-the-node-modules-plugin) and you'll be ready to go in a minute 🙂
```

File: packages/gatsby/content/advanced/migration.md (hunk 2 above)
Comment (paul-soporan) [reply packages/gatsby/content/advanced/migration.md]: I have found [`gatsby-remark-table-of-contents`](https://www.gatsbyjs.org/packages/gatsby-remark-table-of-contents/). I will implement it in another PR.
